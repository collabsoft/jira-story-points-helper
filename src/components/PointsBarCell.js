import React from 'react';
import styled from 'styled-components';

import {
  STATUS_NEW,
  STATUS_INDETERMINATE,
  STATUS_DONE,
  STATUS_BG_COLOR,
  STATUS_BORDER_RADIUS,
  STATUS_FONT_COLOR,
} from './constants';
import i18n from './utils/i18n';
import formatNumber from './utils/formatNumber';

const STATUS_TEXT = {
  [STATUS_NEW]: i18n('txtNew'),
  [STATUS_INDETERMINATE]: i18n('txtIndeterminate'),
  [STATUS_DONE]: i18n('txtDone'),
};

const PointsBarCell = ({
  className,
  points = 0,
  percentage = 0,
  status = STATUS_NEW,
}) => {
  const pntStr = formatNumber(points);

  return (
    <div
      className={className}
      title={`${STATUS_TEXT[status]}: ${pntStr} (${percentage}%)`}
    >
      {pntStr}
    </div>
  );
};

export default styled(PointsBarCell)`
  align-items: center;
  background-color: ${({ status }) => STATUS_BG_COLOR[status]};
  border-radius: ${({ status }) => STATUS_BORDER_RADIUS[status]};
  box-sizing: border-box;
  color: ${({ status }) => STATUS_FONT_COLOR[status]};
  display: flex;
  flex: ${({ percentage }) => percentage};
  justify-content: center;
  line-height: 0.8125rem;
  padding: 4px 6px;
  transition: all 0.3s ease-in-out;
`;
