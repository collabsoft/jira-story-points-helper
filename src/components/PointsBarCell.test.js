import React from 'react';
import { render } from 'enzyme';

import PointsBarCell from './PointsBarCell';

describe('PointsBarCell', function() {
  it('should render without throwing an error', function() {
    expect(render(<PointsBarCell />)).toMatchSnapshot();
  });

  it('should render with props', function() {
    expect(
      render(<PointsBarCell points={5} percentage={0.5} status="new" />)
    ).toMatchSnapshot();
  });
});
