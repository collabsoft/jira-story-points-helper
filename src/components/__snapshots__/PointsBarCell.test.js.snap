// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PointsBarCell should render with props 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "sc-bdVaJa iFLIac",
      "title": "txtNew: 5 (0.5%)",
    },
    "children": Array [
      Object {
        "data": "5",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "title": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "title": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`PointsBarCell should render without throwing an error 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "sc-bdVaJa dhLPFl",
      "title": "txtNew: 0 (0%)",
    },
    "children": Array [
      Object {
        "data": "0",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "title": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "title": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
